lib/ES/Repo.pm

```diff
@@ -151,17 +151,94 @@ sub prepare {
         return $dest;
     }
 
-    local $ENV{GIT_DIR} = $self->git_dir;
+    $self->_extract_from_ref( $dest, $branch, $path );
+    return $dest;
+}
 
-    $dest->mkpath;
-    my $tar = $dest->file('.temp_git_archive.tar');
-    die "File <$tar> already exists" if -e $tar;
-    run qw(git archive --format=tar -o ), $tar, $resolved_branch, $path;
+#===================================
+sub _prepare_sub_dir {
+#===================================
+    my ( $self, $title, $branch, $path, $dest ) = @_;
+    my $source_root = $self->{sub_dirs}->{$branch};
 
-    run "tar", "-x", "-C", $dest, "-f", $tar;
-    $tar->remove;
+    unless ( $self->{keep_hash} ) {
+        $self->_extract_from_dir( $source_root, $dest, $path );
+        return $dest;
+    }
 
-    return $dest;
+    my $has_uncommitted_changes = eval {
+        local $ENV{GIT_WORK_TREE} = $source_root;
+        local $ENV{GIT_DIR} = $ENV{GIT_WORK_TREE} . '/.git';
+        run qw(git diff-index --quiet HEAD --);
+        1;
+    };
+    unless ( $has_uncommitted_changes ) {
```

Comment:
shouldn't it be `if` here ? 
```perl
if ($has_uncommitted_changes ) {
 # execute the code inside ?
}
```

Reply:
Weird! I've got something backwards....

Reply:
OK - here is what happened `--exit-code` causes `run` to throw an exception if there *are* changes. Which causes `eval` to return 0. So, you could say that I named the variable backwards. Fixing.

Reply:
yeah 😄 the name was confusing a little bit.